src/Build/Logging/BaseConsoleLogger.cs

```diff
@@ -29,6 +30,11 @@ internal abstract class BaseConsoleLogger : INodeLogger
         /// </summary>
         private static bool _supportReadingBackgroundColor = true;
 
+        ///<summary>
+        /// propertyOutputMap is the map between a (nodeID and project_context_id) to a target framework
+        ///</summary>
+        protected Dictionary<(int nodeId, int contextId), StringBuilder> propertyOutputMap = new Dictionary<(int nodeId, int contextId), StringBuilder> ();
```

Comment:
I agree, StringBuilder is a temporary helper. It is not good for storage (it is fat for one thing)> It does not belong in collections. Also if you use string, it may be possible to deduplicate (I don't have full context here, but I assume there's few "target framework" strings, so you certainly don't want many copies of the object representing them.